DataFormats/TrackReco/interface/SiPixelTrackProbQXY.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SiPixelTrackProbQXY_H
+#define SiPixelTrackProbQXY_H
+
+#include <cstdint>
+#include <vector>
+
+#include "DataFormats/Common/interface/Association.h"
+#include "DataFormats/Common/interface/RefVector.h"
+
+namespace reco {
+  // Class defining the combined charge and shape probabilities for tracks that go through the SiPixel detector
+  class SiPixelTrackProbQXY {
+  public:
+    SiPixelTrackProbQXY() {}
+
+    SiPixelTrackProbQXY(float probQonTrack, float probXYonTrack, float probQonTrackNoLayer1, float probXYonTrackNoLayer1)
+        : probQonTrack_(probQonTrack),
+          probXYonTrack_(probXYonTrack),
+          probQonTrackNoLayer1_(probQonTrackNoLayer1),
+          probXYonTrackNoLayer1_(probXYonTrackNoLayer1) {}
+
+    // Return the combined charge probabilities for tracks that go through the SiPixel detector
+    float probQonTrack() const { return probQonTrack_; }
+
+    // Return the combined shape probabilities for tracks that go through the SiPixel detector
+    float probXYonTrack() const { return probXYonTrack_; }
+
+    // Return the combined charge probabilities for tracks that go through the SiPixel detector
+    // This version now excludes layer 1 which was known to be noisy for 2017/2018
+    float probQonTrackNoLayer1() const { return probQonTrackNoLayer1_; }
```

Comment:
So you suggest to have a new header
DataFormats/TrackReco/interface/SiPixelTrackProbQXYNoLayer1.h
?

Reply:
no, `probQ` and `probXY` (2 data members are enough).
The code will produce an additional collection instance with `NoLayer1` if really needed.

Reply:
I dont think this is the way we'd like to go, at the analysis level it could be still valuable to compare the `probQ` vs `probQNoLayer1`, so I definitely need to keep 2 variables for a single track. I dont think that's possible in your proposal.

Reply:
It's certainly possible to do it in my proposal. There will be two collections
```C++
auto probQXY = gt2ProbQXY[tkRef];
auto probQXYNoL1 = gt2ProbQXYNoL1[tkRef];
if (probQXY.probQ() > probQXYNoL1.probQ()) cout << "I can compare two values" << endl;
```

Reply:
@slava77 does this mean I have to add a new label?
If I do this
```
      gt2siPixelTrackProbQXY_(
          consumes<reco::SiPixelTrackProbQXYAssociation>(iConfig.getParameter<edm::InputTag>("siPixelTrackProbQXY"))),
      gt2siPixelTrackProbQXYNoLayer1_(
          consumes<reco::SiPixelTrackProbQXYAssociation>(iConfig.getParameter<edm::InputTag>("siPixelTrackProbQXYNoLayer1"))),
```
that would mean a new producer.

So I have to do 
```
      gt2siPixelTrackProbQXY_(
          consumes<reco::SiPixelTrackProbQXYAssociation>(iConfig.getParameter<edm::InputTag>("siPixelTrackProbQXY"))),
      gt2siPixelTrackProbQXYNoLayer1_(
          consumes<reco::SiPixelTrackProbQXYAssociation>(iConfig.getParameter<edm::InputTag>("siPixelTrackProbQXY"))), #--> this one with a label
```
right?

Reply:
that is the idea. Recall though that this is to provide more flexibility on the AOD side for cases where just one or more than two variants may need to be produced.
AODs have longer lifetime and historically we do not aggregate data in larger concrete classes until the content has clearly settled.

Reply:
Hi @slava77 
I get this
```
/afs/cern.ch/user/t/tvami/public/probQInMiniAODMaster/CMSSW_12_2_X_2021-12-02-2300/src/PhysicsTools/PatAlgos/plugins/PATIsolatedTrackProducer.cc: In member function 'virtual void pat::PATIsolatedTrackProducer::produce(edm::Event&, const edm::EventSetup&)':
/afs/cern.ch/user/t/tvami/public/probQInMiniAODMaster/CMSSW_12_2_X_2021-12-02-2300/src/PhysicsTools/PatAlgos/plugins/PATIsolatedTrackProducer.cc:391:93: error: cannot convert 'const value_type' {aka 'const std::vector<reco::SiPixelTrackProbQXY>'} to 'const reco::SiPixelTrackProbQXY*' in initialization
  391 |       const reco::SiPixelTrackProbQXY* siPixelTrackProbQXY = (*gt2siPixelTrackProbQXY)[tkref];
  ```

Any suggestions

Reply:
I can make a commit so you see what I did, but you can expect that it wont even compile like that